src/index.js

```diff
@@ -0,0 +1,157 @@
+export class Database {
+  #id;
+  #data;
+  constructor() {
+    this.#id = 0;
+    this.#data = [];
+  }
+
+  getData() {
+    return structuredClone(this.#data);
+  }
+
+  add(data) {
+    if (typeof data !== "object") throw new Error(`wrong data type provided`);
+
+    this.#id++;
+    data.id = this.#id;
+    this.#data.push(data);
+  }
+
+  remove(id) {
+    this.#data = this.#data.filter((obj) => obj.id !== this.find(id).id);
+  }
+
+  update(object, key, newValue) {
+    object[key] = newValue;
+    return object;
+  }
+
+  find(id) {
+    const result = this.#data.find((obj) => obj.id === id);
+    if (!result) throw new Error(`data with ID ${id} is not found`);
+    return result;
+  }
+}
+
+export class UserDatabase {
+  constructor(data) {
+    this.data = data;
```

Comment:
You should first declare this as a private attribute, otherwise the internal `data` object will be accessible from outside the class